osu.Framework/Graphics/Video/VideoTexture.cs

```
@@ -27,11 +27,14 @@ public VideoTexture(int width, int height, WrapMode wrapModeS = WrapMode.None, W
 
         private NativeMemoryTracker.NativeMemoryLease memoryLease;
 
-        internal override void SetData(ITextureUpload upload, WrapMode wrapModeS, WrapMode wrapModeT)
+        internal override void SetData(ITextureUpload upload, WrapMode wrapModeS, WrapMode wrapModeT, Opacity? uploadOpacity)
         {
             UploadComplete = false;
 
-            base.SetData(upload, wrapModeS, wrapModeT);
+            // We do not support videos with transparency at this point,
+            // so the upload's opacity as well as the texture's opacity
+            // is always opaque.
+            base.SetData(upload, wrapModeS, wrapModeT, Opacity = Opacity.Opaque);
```

should we assert that the incoming opacity is null or opaque, just for safety?

---

That's gonna be very expensive to compute, so I don't think it's a good idea.

---

i just mean
```
if (uploadOpacity != null && uploadOpacity != Opacity.Opaque)
    throw new InvalidOperationException("Video texture uploads must always be opaque");
```

to avoid a user attempting to use this and provide the argument which is being silently ignored.

---

That's fair.